pkg/ruler/ruler_test.go

```diff
@@ -55,10 +57,16 @@ func defaultRulerConfig(store rules.RuleStore) (Config, func()) {
 }
 
 func newTestRuler(t *testing.T, cfg Config) *Ruler {
+	dir, err := ioutil.TempDir("", t.Name())
+	testutil.Ok(t, err)
+	defer os.RemoveAll(dir)
```

Comment:
I think we need to call `os.RemoveAll(dir)` after the test has completed. Similar to the temp dir cleanup returned in the `defaultRulerConfig` function.

Reply:
Fixed this. I've just realized that tracker has no way to stop and close the mmaped file. So with too many tests using the ruler or querier, we will start to run out of file descriptors :-(

Reply:
Fortunately `go test` compiles and run tests for each package separately, so the risk is lower.